stubs/openpyxl/openpyxl/workbook/workbook.pyi

```diff
@@ -20,54 +25,54 @@ class Workbook:
     views: Incomplete
     def __init__(self, write_only: bool = ..., iso_dates: bool = ...) -> None: ...
     @property
-    def epoch(self): ...
+    def epoch(self) -> Any: ...
     @epoch.setter
-    def epoch(self, value) -> None: ...
+    def epoch(self, value: Any) -> None: ...
     @property
-    def read_only(self): ...
+    def read_only(self) -> bool: ...
     @property
-    def data_only(self): ...
+    def data_only(self) -> bool: ...
     @property
-    def write_only(self): ...
+    def write_only(self) -> bool: ...
     @property
-    def excel_base_date(self): ...
+    def excel_base_date(self) -> Any: ...
     @property
-    def active(self): ...
+    def active(self) -> Worksheet: ...
     @active.setter
-    def active(self, value) -> None: ...
-    def create_sheet(self, title: Incomplete | None = ..., index: Incomplete | None = ...): ...
-    def move_sheet(self, sheet, offset: int = ...) -> None: ...
-    def remove(self, worksheet) -> None: ...
-    def remove_sheet(self, worksheet) -> None: ...
-    def create_chartsheet(self, title: Incomplete | None = ..., index: Incomplete | None = ...): ...
-    def get_sheet_by_name(self, name): ...
-    def __contains__(self, key): ...
-    def index(self, worksheet): ...
-    def get_index(self, worksheet): ...
-    def __getitem__(self, key): ...
-    def __delitem__(self, key) -> None: ...
-    def __iter__(self): ...
-    def get_sheet_names(self): ...
+    def active(self, value: Worksheet) -> None: ...
+    def create_sheet(self, title: str | None = ..., index: int | None = ...): ...
+    def move_sheet(self, sheet: Worksheet, offset: int = ...) -> None: ...
+    def remove(self, worksheet: Worksheet) -> None: ...
+    def remove_sheet(self, worksheet: Worksheet) -> None: ...
+    def create_chartsheet(self, title: str | None = ..., index: int | None = ...) -> None: ...
+    def get_sheet_by_name(self, name: str) -> Worksheet: ...
+    def __contains__(self, key: str) -> bool: ...
+    def index(self, worksheet: Worksheet) -> int: ...
+    def get_index(self, worksheet: Worksheet) -> int: ...
+    def __getitem__(self, key: str) -> Worksheet: ...
+    def __delitem__(self, key: str) -> None: ...
+    def __iter__(self) -> Iterator[Worksheet]: ...
+    def get_sheet_names(self) -> list[Worksheet]: ...
     @property
-    def worksheets(self): ...
+    def worksheets(self) -> list[Worksheet]: ...
     @property
-    def chartsheets(self): ...
+    def chartsheets(self) -> list[Chartsheet]: ...
     @property
-    def sheetnames(self): ...
+    def sheetnames(self) -> list[str]: ...
     def create_named_range(
-        self, name, worksheet: Incomplete | None = ..., value: Incomplete | None = ..., scope: Incomplete | None = ...
+        self, name: str, worksheet: Worksheet | None = ..., value: str | Any | None = ..., scope: Incomplete | None = ...
     ) -> None: ...
-    def add_named_style(self, style) -> None: ...
+    def add_named_style(self, style: Any) -> None: ...
```

Comment:
This calls `self._named_styles.append()`: https://foss.heptapod.net/openpyxl/openpyxl/-/blob/branch/3.0/openpyxl/workbook/workbook.py#L344. `self._named_styles` is a `NamedStyleList` object, and `NamedStyleList.append()` raises `TypeError` if you pass it an object that isn't a `NamedStyle`: https://foss.heptapod.net/openpyxl/openpyxl/-/blob/branch/3.0/openpyxl/styles/named_styles.py#L190-191

```suggestion
    def add_named_style(self, style: NamedStyle) -> None: ...
```